File: src/main/java/workshop/part2/UserActor.java
package workshop.part2;

import akka.actor.AbstractActor;
import workshop.common.userservice.UserCriminalRecord;
import workshop.common.userservice.UserService;

public class UserActor extends AbstractActor {

    private final UserService userService;

    public UserActor(UserService userService) {
        this.userService = userService;
    }

    @Override
    public Receive createReceive() {
        return receiveBuilder()
            .build();
    }

    // Reply message type
    public static class CheckUserResult {
        public final UserCriminalRecord record;

        public CheckUserResult(UserCriminalRecord record) {
            this.record = record;
        }
    }

    // Request message type
    public static class CheckUser {
        public final Integer userId;

        public CheckUser(Integer userId) {
            this.userId = userId;
        }
    }
}

File: src/main/kotlin/workshop/part2/UserActor.kt
package workshop.part2

import akka.actor.UntypedActor
import workshop.common.userservice.UserCriminalRecord
import workshop.common.userservice.UserService

class UserActor(private val userService: UserService) : UntypedActor() {

    override fun onReceive(message: Any?) {
        TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
    }

    class CheckUserResult(val record: UserCriminalRecord)

    class CheckUser(val userId: Int?)
}
